var/spack/repos/builtin/packages/seacas/package.py

```diff
@@ -0,0 +1,410 @@
+# Copyright 2013-2018 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import sys
+from spack import *
+from spack.operating_systems.mac_os import macos_version
+
+#
+# Need to add:
+#  Thread enable/disable
+#  Kokkos enable/disable
+#  Data Warehouse (FAODEL) enable/disable
+
+
+class Seacas(CMakePackage):
+    """The SEACAS Project contains the Exodus and IOSS libraries and a
+     collection of applications which create, query, modify, or
+     translate exodus databases.
+    """
+    homepage = "http://gsjaardema.github.io/seacas/"
+    git      = "https://github.com/gsjaardema/seacas.git"
+
+    maintainers = ['gsjaardema']
+
+    # ###################### Versions ##########################
+
+    version('master', branch='master')
+
+    # ###################### Variants ##########################
+
+    # Build options
+    variant('fortran',      default=True,
+            description='Compile with Fortran support')
+    variant('shared',       default=True,
+            description='Enables the build of shared libraries')
+    variant('mpi', default=False, description='Enables MPI parallelism.')
+
+    variant('thread_safe',  default=False,
+            description='Enable thread-safe exodus and IOSS libraries')
+    variant('kokkos',       default=False,
+            description='Compile with Kokkos')
+
+    # TPLs (alphabet order)
+    variant('cgns',         default=True,
+            description='Enable CGNS')
+    variant('matio',        default=True,
+            description='Compile with matio (MatLab) support')
+    variant('metis',        default=False,
+            description='Compile with METIS and ParMETIS')
+    variant('x11',          default=True,
+            description='Compile with X11')
+
+    # Package options
+    variant('allpkgs',   default=True,
+            description='Compile with all packages')
```

Comment:
These kind of variants are always a bit controversial. As far as Spack is concerned, if you select `-allpkgs` and `+every_other_variant`, this will be a different spec than `+allpkgs` and `-every_other_variant`, or `+allpkgs` and `+some_variants`. It would be ideal if there was only one way to specify the same thing. But I can understand why you would want a variant to enable all other variants with this many of them.

Reply:
Yes, I agree.  The majority of the time, I expect that one of the following would be selected:

- All packages will be built
- Only exodus library will be built (may be possible to eliminate exodusII package and just use `seacas+exodus` instead?
- Only exodus and Ioss libraries

Maybe it would be better to just provide a smaller subset of package subsetting options?